lib/spack/spack/test/util/executable.py

```diff
@@ -40,6 +40,36 @@ def test_read_unicode(tmpdir, working_env):
         assert u'\xc3' == script(output=str).strip()
 
 
+def test_which_relative_path_with_slash(tmpdir, working_env):
+    tmpdir.ensure('exe')
+    path = str(tmpdir.join('exe'))
+    os.environ['PATH'] = ''
```

Comment:
Just a comment. I see this is taken care of by `working_env` but in this case using `monkeypatch.setenv` would equally do.